Add Theora support

RetroGamer02 opened the pull request:
Adds support for the Theora video codec as well as build instructions.

Also see [https://github.com/RetroGamer02/libtheora_for_3DS](https://github.com/RetroGamer02/libtheora_for_3DS)

Core-2-Extreme commented:
> Adds support for the Theora video codec as well as build instructions.

It looks we need to have `libogg` when building `libtheora`, did you somehow managed to build it without `libogg` or built `libogg` but forget commit them as well?

RetroGamer02 commented:
Added new build instructions and the following repos.

https://github.com/RetroGamer02/libogg_for_3DS

https://github.com/RetroGamer02/libvorbis_for_3DS

Core-2-Extreme commented:
Thank you for updates, I've managed to build it and successfully played theora videos.

I noticed :
1. We don't need `lbvorbis` to build `libtheora` and since we already have `vorbis` support via `FFmpeg` we don't need it.
    * -> Can you remove it unless you need to keep it for some reasons.
2. There is a commit called [Disable ARM NEON](https://github.com/RetroGamer02/libtheora_for_3DS/commit/4f38b04f391d150e292446982d9afd02e452a487), you've modified configure and added an assembly file (`armloop.s`) here.
    * -> Is this commit supposed to make it able to build with assembly optimizations (i.e. without `--disable-asm`)?
        * (I've built `libtheora` with `--disable-asm` because there was an compilation errors without it.)
    * -> Is `armloop.s` written by you or you just forgot to commit it on previous commit?
3. Some typos, missing links and very minor issue.
    * -> I'll make suggestions for that, but I need to know No.1 and No.2 first so please answer that questions so that I can make suggestions.

RetroGamer02 commented:
> Thank you for updates, I've managed to build it and successfully played theora videos.
> 
> I noticed :
> 
> 1. We don't need `lbvorbis` to build `libtheora` and since we already have `vorbis` support via `FFmpeg` we don't need it.
>    
>    * -> Can you remove it unless you need to keep it for some reasons.
> 2. There is a commit called [Disable ARM NEON](https://github.com/RetroGamer02/libtheora_for_3DS/commit/4f38b04f391d150e292446982d9afd02e452a487), you've modified configure and added an assembly file (`armloop.s`) here.
>    
>    * -> Is this commit supposed to make it able to build with assembly optimizations (i.e. without `--disable-asm`)?
>      
>      * (I've built `libtheora` with `--disable-asm` because there was an compilation errors without it.)
>    * -> Is `armloop.s` written by you or you just forgot to commit it on previous commit?
> 3. Some typos, missing links and very minor issue.
>    
>    * -> I'll make suggestions for that, but I need to know No.1 and No.2 first so please answer that questions so that I can make suggestions.

I can try removing libvorbis sure. Also I disabled NEON so it could build with ARM ASM Optimizations yes. As for armloop.s It seems the devs forgot to package it so I added it manually. See here https://gitlab.xiph.org/xiph/theora/-/commit/b167cb4a3e3e2d3239d71f0ca2cbc25af4f2183c

Sorry about the typo's if you can tell me whats wrong ill fix it asap.

Core-2-Extreme commented:
> Also I disabled NEON so it could build with ARM ASM Optimizations yes.
> As for armloop.s It seems the devs forgot to package it so I added it manually. See here https://gitlab.xiph.org/xiph/theora/-/commit/b167cb4a3e3e2d3239d71f0ca2cbc25af4f2183c

Ok, I could build it with assembly optimization after adding `D__3DS__` (from your last commit).
I see your `libtheora_for_3DS` and noticed you've just uploaded original repository so I can't see commit logs before your upload.
So can you ***fork*** (branch name can be anything) [original repository](https://github.com/xiph/theora/tree/master) and apply your changes (disable NEON)?
Also, it looks original repository contains `armloop.s` so you don't have to upload it yourself.

Core-2-Extreme commented:
After merging this PR, I'll:

* Rebuild with `devkitpro r64-2` because other libraries are still using `r64-2` so I want to keep the compiler version the same.
* Add optional instruction to remove `libtheoraenc` manually because we don't use it.

Core-2-Extreme commented:
Thank you for your fixing
